### ISSUE TITLE
Client Pong timed out

### ISSUE THREAD

**Issue by rachelxj** [SocketCluster/socketcluster-client-ios#4]
Can't communiate with server, client pong timed out, the connection is disconnected too often

**Comment by lihan** [comment]
@rachelxj @jondubois  Can you give detail about how we can reproduce this.

**Comment by elondon** [comment]
Hi @lihan -

Just FYI - I'm experiencing the same thing with a very basic setup. I created a socketcluster app, fired it up with node server.js. I installed socketcluster-client-iOS via cocoapods and followed the instructions exactly. 

Client side gets a lot of this:

WVJB RCVD: {"callbackId":"cb_6_1456874992151","handlerName":"onErrorHandler","data":{"data":"Server ping timed out"}}
WVJB RCVD: {"callbackId":"cb_9_1456875013152","handlerName":"onErrorHandler","data":{"data":"Client pong timed out"}}

Server side gets a lot of this:

 [Warning] SocketProtocolError: Client pong timed out

And eventually this:

[Warning] SocketProtocolError: Socket hung up

It appears to be reconnecting after the time out - but then just times out again. That's why I'm seeing several of these. 

Is this library actively developed and tested?

**Comment by jondubois** [comment]
@elondon Yes this client out of date (that's what's causing the issue)... I asked @lihan if he can update it so stay tuned.

**Comment by ricky1550** [comment]
@lihan [Warning] SocketProtocolError: Socket hung up  is this issue fixed?

**Comment by ricky1550** [comment]
@lihan @elondon Please respond if this library is still  supported by you.
